firmware/controllers/limp_manager.cpp

```diff
@@ -148,11 +148,32 @@ void LimpManager::updateState(int rpm, efitick_t nowNt) {
 		allowFuel.clear(ClearReason::StopRequested);
 	}
 
-	// If duty cycle is high, impose a fuel cut rev limiter.
-	// This is safer than attempting to limp along with injectors or a pump that are out of flow.
-	// only reset once below 20% duty to force the driver to lift
-	if (m_injectorDutyCutHysteresis.test(getInjectorDutyCycle(rpm), 96, 20)) {
-		allowFuel.clear(ClearReason::InjectorDutyCycle);
+	{
+		// If duty cycle is high, impose a fuel cut rev limiter.
```

Comment:
does scheduling code support 100% duty cycle?

Reply:
I believe it can, because of the pooled `scheduling_s` used for injection scheduling. Multiple injections get scheduled over top of one another, and the overlapping injection counters prevent closing the injector early.

I need to double check this on real hardware, though.